Documentation/json-schema/distro-support-schema.json

```diff
@@ -0,0 +1,120 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "type": "object",
+    "properties": {
+      "DISTRO": {
+        "description": "The Distro name as in cat etc/os-release. Multiple distros are added to the same file.",
+        "type": "object",
+        "properties":
+        {
+            "installCommand": {

```

Comment:
I'd really suggest doing a string[] for commands here, and make sudo execution a boolean flag for each command. Without a string array, you're going to have to split and parse command lines per-token, and that gets very hairy very quickly. not to mention quoted arguments or multi-word arguments. trust me that moving to an array will make your life easier.

consider a signature like this for each command here:

```typescript
type Command = string[];
type FullCommand = { sudo: boolean; singleCommand: Command} | { sudo: boolean; compoundCommands: Command[] }
```

then in code you can pass the string[] of tokens to spawn/spawnSync, and if you need to orchestrate compoundcommands you can just concat the arrays for each sub-Command together, joining them with a "&&" token between.